lib/teleterm/gateway/app.go

```diff
@@ -55,11 +54,29 @@ func makeAppGateway(cfg Config) (Gateway, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	middleware := &appMiddleware{
+		log: a.cfg.Log,
+		onExpiredCert: func(ctx context.Context) error {
+			_, err := a.cfg.OnExpiredCert(ctx, a)
+			return trace.Wrap(err)
```

Comment:
Is it possible to set the cert from here directly?
```suggestion
			cert, err := a.cfg.OnExpiredCert(ctx, a)
			if err != nil {
				return trace.Wrap(err)
			}
			if a.localProxy == nil { // Shouldn't happen
			...
			}
			a.localProxy.SetCerts([]tls.Certificate{cert})
			return nil
```

Reply:
I refactored the code so that the middleware calls `onExpiredCert` to get a fresh cert and then sets the cert on the local proxy.

I thought about it when working on this PR, but initially I decided set the cert in `clusters.Cluster.ReissueGatewayCerts` because that's what we do for db gateways as well.

Ultimately, it's better to assume that what db gateways do is wrong and move the responsibility behind setting the cert to the middleware itself. That's what the kube middleware does, as well as the db middleware used in tsh – which is the one that's going to be used in lib/teleterm in the future as well.

So once we get rid of the current db middleware, all local proxy middlewares are going to set the certs on underlying proxies by themselves.